File: projects/GPL-FH/src/UndirectedWithEdges/GPL/Vertex.java
package GPL;

import java.util.Iterator;
import java.util.LinkedList;

// *********************************************************************

public class Vertex 
{
    public LinkedList neighbors;
    public String name;

    public Vertex( ) 
    {
        VertexConstructor( );
    }

    public void VertexConstructor( ) 
    {
        name      = null;
        neighbors = new LinkedList( );
    }

    public  Vertex assignName( String name ) 
    {
        this.name = name;
        return ( Vertex ) this;
    }

    public String getName( )
    {
        return this.name;
    }

    public LinkedList getNeighborsObj( )
    {
 	  return neighbors;
    }


    public VertexIter getNeighbors( )
    {
        return new VertexIter( )
        {
            private Iterator iter = neighbors.iterator( );
            public Vertex next( ) 
            { 
              return ( ( Neighbor )iter.next( ) ).end; 
            }
            public boolean hasNext( ) 
            { 
              return iter.hasNext( ); 
            }
        };
    }

    public void display( ) 
    {
        System.out.print( " Node " + name + " connected to: " );

        for ( VertexIter vxiter = getNeighbors( ); vxiter.hasNext( ); )
        {
            System.out.print( vxiter.next().getName() + ", " );
        }

        System.out.println( );
    }      
//--------------------
// differences
//--------------------

    public void addNeighbor( Neighbor n ) 
    {
        neighbors.add( n );
    }

    public EdgeIter getEdges( )
    {
        return new EdgeIter( )
        {
            private Iterator iter = neighbors.iterator( );
            public EdgeIfc next( ) 
            { 
              return ( ( EdgeIfc ) ( ( Neighbor )iter.next( ) ).edge );
            }
            public boolean hasNext( ) 
            { 
              return iter.hasNext( ); 
            }
        };
    }

}

File: projects/GPL-FH-CSharp/features/UndirectedWithEdges/Vertex.cs
//layer Undirected;

//import java.util.Iterator;
using System.Collections;
using System.Collections.Generic;

// *********************************************************************

public class Vertex 
{
    public List<Neighbor> neighbors;
    public string name;

    public Vertex( ) 
    {
        VertexConstructor( );
    }

    public void VertexConstructor( ) 
    {
        name      = null;
        neighbors = new List<Neighbor>( );
    }

    public  Vertex assignName( string name ) 
    {
        this.name = name;
        return ( Vertex ) this;
    }

    public string GetName( )
    {
        return this.name;
    }

    public List<Neighbor> GetNeighborsObj( )
    {
 	  return neighbors;
    }


    public Iterator<Vertex> GetNeighbors() {
        return new Iterator<Vertex, Neighbor>(neighbors.GetEnumerator(), TransformNeighborToEnd);
    }
    public static Vertex TransformNeighborToEnd(Neighbor val) {
        // should be a Lambda Method in C# 3.0
        return val.end;
    }
    public void display( ) 
    {
        System.Console.Out.Write( " Node " + name + " connected to: " );

        for ( Iterator<Vertex> vxiter = GetNeighbors( ); vxiter.hasNext( ); )
        {
            System.Console.Out.Write( vxiter.next().GetName() + ", " );
        }

        System.Console.Out.WriteLine( );
    }      
//--------------------
// differences
//--------------------

    public void AddNeighbor( Neighbor n ) 
    {
        neighbors.Add( n );
    }

        public Iterator<EdgeIfc> GetEdges() {
        return new Iterator<EdgeIfc, Neighbor>(neighbors.GetEnumerator(), TransformNeighborToEdge);
    }
    public static EdgeIfc TransformNeighborToEdge(Neighbor val) {
        // should be a Lambda Method in C# 3.0
        return val.edge;
    }
}
